### PR TITLE
Add image accessor to MGLStyle

### PR DESCRIPTION
**Pull request by rmnblm**
@1ec5: I'm struggling a bit right now. Could you give a quick review for the current state of implementation to see if I'm heading into the right direction? What's currently not working is to initialize an MGLImage from the byte array of a sprite image.

This PR resolves issue #7012.

### CODE REVIEW

**Comment by rmnblm** [comment]
@1ec5 Just a small reminder ⏱ (I'm pretty sure that you're totally busy and that's okay, just let me know if you need more details or something I can do ☺️)

**Comment by 1ec5** [comment]
Hi @rmnblm, this is on my plate to take a look at. Indeed, it’s a little busy with the holiday season and preparations for the v3.4.0 release.

Incidentally, I think it would be nice to get this fix into v3.4.x given that it rounds out the runtime styling API. For that to happen, please retarget this PR at the release-ios-v3.4.0 branch, where we’re developing v3.4.0 and subsequent patch releases.

File: platform/darwin/src/MGLStyle.mm [hunk 1]
```diff
@@ -287,6 +287,22 @@ - (void)removeImageForName:(NSString *)name
...
+
+    auto spriteImage = self.mapView.mbglMap->getImage([name UTF8String]);
+
+    NSAssert(spriteImage, @"spriteImage is null");
+    if (!spriteImage)
+    {
+        return nil;
+    }
+
+    auto imageData = [NSData dataWithBytes:spriteImage->image.data.get() length:spriteImage->image.bytes()];
```
Comment (1ec5) [inline platform/darwin/src/MGLStyle.mm]: You probably want to `PremultipliedImage` has a field called `data` that likely has what you want.

**Comment by rmnblm** [comment]
@1ec5 I allowed edits from maintainers

**Comment by 1ec5** [comment]
Thanks for your contribution, @rmnblm!

**Comment by 1ec5** [comment]
9456bec1721f5cd69e7813c68b829db4c7ff9660 fixes an mbgl test failure caused by this PR.

For future reference, if you make changes to mbgl or its tests, please run `make xproj` and run the mbgl-test scheme. (You can also turn on the `--gtest-filter=` argument that’s currently disabled in the scheme editor to run a specific test.) Unfortunately, we don’t run CI builds for third-party PRs. I’ll try to remember to push a dry-run branch in the main repository, to make sure all the CI builds pass, whenever we get one.

**Comment by rmnblm** [comment]
Thanks for merging and especially fixing the subsequent issue @1ec5.
Like the title of the PR said it was WIP but it surely wasn't clever from me to check in a currently failing test so it was clearly my fault, sorry.

**Comment by 1ec5** [comment]
No worries. Looking forward to your future PRs! 😃

**Pull request merged**